Add readme content (#4)
DevExpress-Examples/blazor-use-devextreme-circular-gauge: README.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T1069428)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->
# Blazor - Use Gauges, HTML Editor, and Map in a Blazor Application

> Blazor applications allow you to use external JavaScript libraries to integrate components that are not currently present in the DevExpress Blazor suite. This example demonstrates how to incorporate [DevExtreme widgets](https://js.devexpress.com/Demos/WidgetsGallery/) into a Blazor application. 

Reviewer: ```suggestion
This article shows how you can do embed [DevExtreme widgets](https://js.devexpress.com/Demos/WidgetsGallery/) into your Blazor application. You can use the same methods to integrate any JavaScript-based libraries.
```

Author: ```suggestion
This example shows how you can embed [DevExtreme widgets](https://js.devexpress.com/Demos/WidgetsGallery/) into your Blazor application. You can use the same methods to integrate any JavaScript-based libraries.

```

![HTML Editor in DevExpress Blazor App](readme.md)

## Implementation Details

The example solution contains **JSWidgetsRCL** and **DxtComponentsInBlazor** projects. The first project wraps the following DevExtreme widgets:

* [Gauge](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Gauges/Overview/jQuery/Light/)
* [HTML Editor](https://js.devexpress.com/Demos/WidgetsGallery/Demo/HtmlEditor/Overview/jQuery/Light/)
* [Map](https://js.devexpress.com/Demos/WidgetsGallery/Demo/VectorMap/Overview/jQuery/Light/)

> For each widget, _DevExtreme<WidgetName>.razor_ and _DevExtreme<WidgetName>.razor.js_ files store wrapper implementation. HTML Editor also implements a custom event that handles changes in the component's content: registers a [custom event handler](/JSWidgetsRCL/EventHandlers.cs), implements it on a [Blazor side](/JSWidgetsRCL/DevExtremeComponents/DevExtremeHtmlEditor.razor#L21) and maps it to the corresponding [DevExtreme event handler](/JSWidgetsRCL/DevExtremeComponents/DevExtremeHtmlEditor.razor.js#L7). You can implement other events in the same way if needed.

Reviewer: First of all, the first and second sentence weren't really flowing from one to the other. One talks about two files. And then the other just continues with something else... Do I need to look in one of the files? Which one? 

But please check what I wrote - I may have messed up the file name. 

```suggestion
For each widget, _DevExtreme<WidgetName>.razor_ and _DevExtreme<WidgetName>.razor.js_ files store wrapper implementation. Look at _DevExtremeHtmlEditor.razor_ for an example on how to implement an event. The wrapper registers a [custom event handler](/JSWidgetsRCL/EventHandlers.cs), implements it on the [Blazor side](/JSWidgetsRCL/DevExtremeComponents/DevExtremeHtmlEditor.razor#L21) and maps it to the corresponding [DevExtreme event handler](/JSWidgetsRCL/DevExtremeComponents/DevExtremeHtmlEditor.razor.js#L7). You can implement other events in the same way.
```

Each widget also renders the **DevExtremeResources.razor** component. This component loads DevExtreme resources when you open a page with a DevExtreme component for the first time.

> The **DxtComponentsInBlazor** project references **JSWidgetsRCL**. This allows you to use wrappers as regular Blazor components, for instance:

Reviewer: ```suggestion
The **DxtComponentsInBlazor** project references **JSWidgetsRCL**. This allows you to use wrappers as regular Blazor components. The following code adds a `DevExtremeMap` wrapper component:

```
